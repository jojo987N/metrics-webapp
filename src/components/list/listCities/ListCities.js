import { useSelector } from 'react-redux';
import './listCities.css';
import City from '../../city/City';

export default function ListCities() {
  const { data1, isLoading, error } = useSelector((state) => state.cities);
  if (isLoading) return <div className="loading">Loading ...</div>;
  if (error) { return <div className="error">API doesn&apos;t provide data for this country</div>; }
  return (
    <div className="container">
      {data1.map((item, i) => <City key={item.city} item={item} i={i} />)}
    </div>
  );
}
