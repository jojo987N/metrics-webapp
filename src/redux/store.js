import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';
import citiesReducer from './cities/citiesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    cities: citiesReducer,
  },
});
export default store;
