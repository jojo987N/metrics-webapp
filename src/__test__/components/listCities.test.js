import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../__mock__/storeMock';
import ListCities from '../../components/list/listCities/ListCities';

it('renders correctly', () => {
  const tree = render(
    <Provider store={store}>
      <ListCities />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
