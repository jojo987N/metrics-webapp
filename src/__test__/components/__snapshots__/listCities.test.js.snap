// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="container-city dark1"
        >
          <h1
            class="title"
          >
            ZAGREB
          </h1>
          <p
            class="persons"
          >
            790,017
             
            people
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="container-city dark"
        >
          <h1
            class="title"
          >
            Split
          </h1>
          <p
            class="persons"
          >
            178,102
             
            people
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="container-city dark1"
        >
          <h1
            class="title"
          >
            Rijeka
          </h1>
          <p
            class="persons"
          >
            128,624
             
            people
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="container-city dark"
        >
          <h1
            class="title"
          >
            Osijek
          </h1>
          <p
            class="persons"
          >
            108,048
             
            people
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="container-city dark1"
      >
        <h1
          class="title"
        >
          ZAGREB
        </h1>
        <p
          class="persons"
        >
          790,017
           
          people
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
          data-icon="circle-arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="container-city dark"
      >
        <h1
          class="title"
        >
          Split
        </h1>
        <p
          class="persons"
        >
          178,102
           
          people
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
          data-icon="circle-arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="container-city dark1"
      >
        <h1
          class="title"
        >
          Rijeka
        </h1>
        <p
          class="persons"
        >
          128,624
           
          people
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
          data-icon="circle-arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="container-city dark"
      >
        <h1
          class="title"
        >
          Osijek
        </h1>
        <p
          class="persons"
        >
          108,048
           
          people
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-arrow-right faCircleArrowRight"
          data-icon="circle-arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
